java/memory/src/main/java/org/apache/arrow/memory/util/hash/NonNegativeHasher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.memory.util.hash;
+
+/**
+ * Wrapper for another {@link ArrowBufHasher}, to make the generated
+ * hash code non-negative.
+ */
+public class NonNegativeHasher extends ArrowBufHasher {
+
+  private ArrowBufHasher innerHasher;
+
+  /**
+   * Constructs a non-negative wrapper for another {@link ArrowBufHasher}.
+   * @param innerHasher the hasher to be wrapped.
+   */
+  public NonNegativeHasher(ArrowBufHasher innerHasher) {
+    this.innerHasher = innerHasher;
+  }
+
+  @Override
```

Comment:
this type of overriding makes me think there should probably be an interface someplace to implement instead of relying on abstract classes.

Reply:
Would you please elaborate your thoughts in more details?

Reply:
It seems strange to override and just delegate these methods.  This could be avoided which an interface exposing only the public method currently exposed on the class.

Reply:
also it seems the protected methods aren't necessarily useful for implementing other hash functions (like murmurhash, etc)